test/accel/TrackingManagerIntegration.test.cc

```diff
@@ -409,6 +415,273 @@ TEST_F(LarSphereOptical, run)
     rm.BeamOn(2);
 }
 
+//---------------------------------------------------------------------------//
+// OpNovice
+//---------------------------------------------------------------------------//
+class OpNovice : public OpNoviceIntegrationMixin, public TMITestBase
+{
+    void BeginOfEventAction(G4Event const* event) override
+    {
+        if (event->GetEventID() == 1)
+        {
+            for (auto i : range(event->GetNumberOfPrimaryVertex()))
+            {
+                G4PrimaryVertex* vtx = event->GetPrimaryVertex(i);
+                for (auto j : range(vtx->GetNumberOfParticle()))
+                {
+                    G4PrimaryParticle* p = vtx->GetPrimary(j);
+                    p->SetWeight(10.0);
+                }
+            }
+        }
+    }
```

Comment:
There's no need to adjust the weight of the particles; this was done once to test that weight gets correctly sent to the hit objects.